sdk/eventhub/azure-eventhub-checkpointstoretable/azure/eventhub/extensions/checkpointstoretable/_tablestoragecs.py

```diff
@@ -17,22 +56,299 @@ class TableCheckpointStore:
+                  relative to the Event Hubs namespace that contains it.
+                - `consumer_group` (str): The name of the consumer group the checkpoints are associated with.
+                - `partition_id` (str): The partition ID which the checkpoint is created for.
+                - `sequence_number` (int): The sequence number of the :class:`EventData<azure.eventhub.EventData>`.
+                - `offset` (str): The offset of the :class:`EventData<azure.eventhub.EventData>`.
+        """
+        partition_key = "{} {} {} Checkpoint".format(
+            fully_qualified_namespace, eventhub_name, consumer_group
+        )
+        partition_key_filter = "PartitionKey eq '{}'".format(partition_key)
+        entities = self._table_client.query_entities(partition_key_filter, **kwargs)
+        checkpoints_list = []
+        for entity in entities:
+            checkpoint = {
+                "fully_qualified_namespace": fully_qualified_namespace,
+                "eventhub_name": eventhub_name,
+                "consumer_group": consumer_group,
+                "partition_id": entity[u"RowKey"],
+                "sequence_number": entity[u"sequencenumber"],
+                "offset": str(entity[u"offset"]),
+            }
+            checkpoints_list.append(checkpoint)
+        return checkpoints_list
 
     def update_checkpoint(self, checkpoint, **kwargs):
-        pass
+        """Updates the checkpoint using the given information for the offset, associated partition and
+        consumer group in the storage table.
+
+        Note: If you plan to implement a custom checkpoint store with the intention of running between
+        cross-language EventHubs SDKs, it is recommended to persist the offset value as an integer.
+        :param Dict[str,Any] checkpoint: A dict containing checkpoint information:
+                - `fully_qualified_namespace` (str): The fully qualified namespace that the Event Hub belongs to.
+                  The format is like "<namespace>.servicebus.windows.net".
+                - `eventhub_name` (str): The name of the specific Event Hub the checkpoint is associated with,
+                  relative to the Event Hubs namespace that contains it.
+                - `consumer_group` (str): The name of the consumer group the checkpoint is associated with.
+                - `partition_id` (str): The partition ID which the checkpoint is created for.
+                - `sequence_number` (int): The sequence number of the :class:`EventData<azure.eventhub.EventData>`
+                  the new checkpoint will be associated with.
+                - `offset` (str): The offset of the :class:`EventData<azure.eventhub.EventData>`
+                  the new checkpoint will be associated with.
+        :rtype: None
+        """
+        try:
+            checkpoint_entity = TableCheckpointStore._create_checkpoint_entity(
+                checkpoint
+            )
```

Comment:
I think this should be outside the try since it's used in the except too?
```suggestion
        checkpoint_entity = TableCheckpointStore._create_checkpoint_entity(
            checkpoint
        )
        entity_name = "{}/{}/{}/checkpoint/{}".format(
            checkpoint_entity["fully_qualified_namespace"],
            checkpoint_entity["eventhub_name"],
            checkpoint_entity["consumer_group"],
            checkpoint_entity["partition_id"],
        )
        try:
```